pkgs/tools/package-management/pacman/default.nix

```diff
@@ -1,40 +1,86 @@
-{ stdenv, lib, fetchurl, pkg-config, m4, perl, libarchive, openssl, zlib, bzip2,
-xz, curl, runtimeShell }:
+{ lib
+, stdenv
+, fetchurl
+, asciidoc
+, bzip2
+, coreutils
+, curl
+, gpgme
+, installShellFiles
+, libarchive
+, meson
+, ninja
+, openssl
+, perl
+, pkg-config
+, runtimeShell
+, xz
+, zlib
+}:
 
 stdenv.mkDerivation rec {
   pname = "pacman";
-  version = "5.2.2";
+  version = "6.0.1";
 
   src = fetchurl {
-    url = "https://sources.archlinux.org/other/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "1829jcc300fxidr3cahx5kpnxkpg500daqgn2782hg5m5ygil85v";
+    url = "https://sources.archlinux.org/other/${pname}/${pname}-${version}.tar.xz";
+    hash = "sha256-DbYUVuVqpJ4mDokcCwJb4hAxnmKxVSHynT6TsA079zE=";
   };
 
-  enableParallelBuilding = true;
+  nativeBuildInputs = [
+    asciidoc
+    installShellFiles
+    meson
+    ninja
+    pkg-config
+  ];
 
-  configureFlags = [
-    # trying to build docs fails with a2x errors, unable to fix through asciidoc
-    "--disable-doc"
+  buildInputs = [
+    bzip2
+    curl
+    gpgme
+    libarchive
+    openssl
+    perl
+    xz
+    zlib
+  ];
 
-    "--sysconfdir=/etc"
-    "--localstatedir=/var"
-    "--with-scriptlet-shell=${runtimeShell}"
+  hardeningDisable = [
+    "fortify"
   ];
 
-  installFlags = [ "sysconfdir=${placeholder "out"}/etc" ];
+  postPatch = ''
+    substituteInPlace meson.build \
+      --replace "install_dir : SYSCONFDIR" "install_dir : '${placeholder "out"}/etc'" \
+      --replace "join_paths(LOCALSTATEDIR, 'lib/pacman/')," "'${placeholder "out"}/var/lib/pacman/'," \
+      --replace "join_paths(LOCALSTATEDIR, 'cache/pacman/pkg/')," "'${placeholder "out"}/var/cache/pacman/pkg/',"
+    substituteInPlace doc/meson.build \
+      --replace "/bin/true" "${coreutils}/bin/true"
+    substituteInPlace scripts/repo-add.sh.in \
+      --replace bsdtar "${libarchive}/bin/bsdtar"
 
-  nativeBuildInputs = [ pkg-config m4 ];
-  buildInputs = [ curl perl libarchive openssl zlib bzip2 xz ];
+    # workaround for gnupg 2.3.x, as Arch is still using 2.2.x
+    substituteInPlace scripts/pacman-key.sh.in \
+      --replace "--batch --check-trustdb" "--batch --check-trustdb --allow-weak-key-signatures"
+  '';
 
-  postFixup = ''
-    substituteInPlace $out/bin/repo-add \
-      --replace bsdtar "${libarchive}/bin/bsdtar"
+  mesonFlags = [
+    "--sysconfdir=/etc"
+    "--localstatedir=/var"
```

Comment:
Why the localstatedir set here is not applied to the lines above.

Reply:
According to https://github.com/NixOS/nixpkgs/pull/161115:

> It takes some patching of meson.build to have the same behaviour as pacman 5.x : configure the tools to use /etc and /var by default but install files under $out/etc and $out/var.